prometheus/vec.go

```diff
@@ -25,10 +25,27 @@ import (
 // provided in this package.
 type MetricVec struct {
 	mtx      sync.RWMutex // Protects the children.
-	children map[uint64]Metric
+	children map[uint64][]metricLabelValue
 	desc     *Desc
 
 	newMetric func(labelValues ...string) Metric
+	hashAdd   func(h uint64, s string) uint64 // replace hash function for testing collision handling
```

Comment:
Would be cool to find a real collision and use that one. Then we could avoid the plumbing.

In any case a comment nit: Please use English capitalization and punctuation, i.e. start with a capital letter and end with a period.